Sources/GRPCHTTP2TransportNIOPosix/NIOSSL+GRPC.swift

```diff
@@ -144,5 +144,25 @@ extension TLSConfiguration {
 
     self = tlsConfiguration
   }
+
+  @available(macOS 15.0, iOS 18.0, watchOS 11.0, tvOS 18.0, visionOS 2.0, *)
+  init(_ tlsConfig: HTTP2ClientTransport.Posix.Config.TLS) throws {
+    var tlsConfiguration = TLSConfiguration.makeClientConfiguration()
+    tlsConfiguration.certificateChain = try tlsConfig.certificateChain.sslCertificateSources()
+
+    if let privateKey = tlsConfig.privateKey {
+      let privateKeySource = try NIOSSLPrivateKey(privateKey: privateKey)
+      tlsConfiguration.privateKey = .privateKey(privateKeySource)
+    }
+
+    tlsConfiguration.minimumTLSVersion = .tlsv12
+    tlsConfiguration.certificateVerification = CertificateVerification(
+      tlsConfig.serverCertificateVerification
+    )
+    tlsConfiguration.trustRoots = try NIOSSLTrustRoots(tlsConfig.trustRoots)
+    tlsConfiguration.applicationProtocols = ["grpc-exp", "h2"]
+
+    self = tlsConfiguration
```

Comment:
(nit) seems a bit indirect, why not `self = TLSConfiguration.makeClientConfiguration()` then modify self directly?

Reply:
Yeah, not any reason specifically. I wrote the `init` above like this and I just copied this one. I'll modify both.